File: uri/1 - Basics/2867/Main.c
#include <stdio.h>
#include <math.h>
#include <string.h>

int main() {
	unsigned long nl, n, m;
	unsigned long f;

	scanf("%d", &nl);
	for(int i = 0; i < nl; i++) {
		scanf("%ld %ld", &n, &m);
		f = pow(n, m);
		printf("%d\n", get_int_len(f));
	}
	return 0;
}

unsigned long get_int_len (unsigned long value){
  unsigned long l=1;
  while(value>9){ l++; value/=10; }
  return l;
}

File: uri/1 - Basics/2867/Main.go
package main

import (
	"fmt"
	"math/big"
)

func main() {
	var n, m, nl int
	var l *big.Int

	fmt.Scanf("%d", &nl)
	for i := 0; i < nl; i++ {
		fmt.Scanf("%d %d", &n, &m)
		l = powBig(n, m)
		fmt.Printf("%v\n", len(l.String()))
	}

}

func powBig(a, n int) *big.Int {
	tmp := big.NewInt(int64(a))
	res := big.NewInt(1)
	for n > 0 {
		temp := new(big.Int)
		if n%2 == 1 {
			temp.Mul(res, tmp)
			res = temp
		}
		temp = new(big.Int)
		temp.Mul(tmp, tmp)
		tmp = temp
		n /= 2
	}
	return res
}
